src/_gettsim/arbeitslosengeld_2/regelbedarf.py

```diff
@@ -1,6 +1,171 @@
+
+
+@dataclass(frozen=True)
+class RegelsatzAnteilErwachsen:
+    je_erwachsener_bei_zwei_erwachsenen: float
+    je_erwachsener_ab_drei_erwachsene: float
+
+
+@dataclass(frozen=True)
+class RegelsatzAnteilKind:
+    anteil: float
+    min_alter: int
+    max_alter: int
+
+
+@dataclass(frozen=True)
+class RegelsatzAnteilKindNachAlter:
+    bis_5_jahre: RegelsatzAnteilKind
+    zwischen_6_und_13_jahre: RegelsatzAnteilKind
+    zwischen_14_und_24_jahre: RegelsatzAnteilKind
+
+
+@dataclass(frozen=True)
+class RegelsatzAnteilsbasiert:
+    basissatz: float
+    erwachsen: RegelsatzAnteilErwachsen
+    kind: RegelsatzAnteilKindNachAlter
+
+
+@params_function(start_date="2005-01-01", end_date="2010-12-31")
+def regelsatz_anteilsbasiert(
+    parameter_regelsatz_anteilsbasiert: RawParamsRequiringConversion,
+) -> RegelsatzAnteilsbasiert:
+    """Regelsatz as a fraction of the Basissatz."""
+    anteilssätze_kinder = parameter_regelsatz_anteilsbasiert[
+        "anteil_vom_basissatz_für_kinder"
+    ]
+    kind_bis_5_jahre = RegelsatzAnteilKind(
+        anteil=anteilssätze_kinder["bis_5_jahre"]["anteil"],
+        min_alter=anteilssätze_kinder["bis_5_jahre"]["min_alter"],
+        max_alter=anteilssätze_kinder["bis_5_jahre"]["max_alter"],
+    )
+    kind_zwischen_6_und_13_jahre = RegelsatzAnteilKind(
+        anteil=anteilssätze_kinder["zwischen_6_und_13_jahre"]["anteil"],
+        min_alter=anteilssätze_kinder["zwischen_6_und_13_jahre"]["min_alter"],
+        max_alter=anteilssätze_kinder["zwischen_6_und_13_jahre"]["max_alter"],
+    )
+    kind_zwischen_14_und_24_jahre = RegelsatzAnteilKind(
+        anteil=anteilssätze_kinder["zwischen_14_und_24_jahre"]["anteil"],
+        min_alter=anteilssätze_kinder["zwischen_14_und_24_jahre"]["min_alter"],
+        max_alter=anteilssätze_kinder["zwischen_14_und_24_jahre"]["max_alter"],
+    )
+    erwachsen = RegelsatzAnteilErwachsen(
+        je_erwachsener_bei_zwei_erwachsenen=parameter_regelsatz_anteilsbasiert[
+            "anteil_vom_basissatz_bei_zwei_erwachsenen"
+        ],
+        je_erwachsener_ab_drei_erwachsene=parameter_regelsatz_anteilsbasiert[
+            "anteil_vom_basissatz_bei_weiteren_erwachsenen"
+        ],
+    )
+    return RegelsatzAnteilsbasiert(
+        basissatz=parameter_regelsatz_anteilsbasiert["basissatz"],
+        erwachsen=erwachsen,
+        kind=RegelsatzAnteilKindNachAlter(
+            bis_5_jahre=kind_bis_5_jahre,
+            zwischen_6_und_13_jahre=kind_zwischen_6_und_13_jahre,
+            zwischen_14_und_24_jahre=kind_zwischen_14_und_24_jahre,
+        ),
+    )
+
+
+@dataclass(frozen=True)
+class RegelbedarfsstufeErwachsener:
+    regelsatz: float
+
+
+@dataclass(frozen=True)
+class RegelbedarfsstufeKind:
+    regelsatz: float
+    min_alter: int
+    max_alter: int
+
+
+@dataclass(frozen=True)
+class RegelsatzNachRegelbedarfsstufen:
+    """Regelsatz as a fraction of the Basissatz."""
+
+    regelbedarfsstufe_1: RegelbedarfsstufeErwachsener
```

Comment:
```suggestion
    rbs_1: RegelbedarfsstufeErwachsener
```
Should be fine here, after all we are only working around not being able to do `1: ...` directly.